nova_vm/src/ecmascript/abstract_operations/type_conversion.rs

```diff
@@ -829,15 +829,34 @@ pub(crate) fn to_big_int(
 }
 
 /// ### [7.1.14 StringToBigInt ( str )](https://tc39.es/ecma262/#sec-stringtobigint)
-pub(crate) fn string_to_big_int(_agent: &mut Agent, _argument: String) -> Option<BigInt> {
+pub(crate) fn string_to_big_int(agent: &mut Agent, argument: String) -> Option<BigInt> {
     // 1. Let text be StringToCodePoints(str).
     // 2. Let literal be ParseText(text, StringIntegerLiteral).
     // 3. If literal is a List of errors, return undefined.
     // 4. Let mv be the MV of literal.
     // 5. Assert: mv is an integer.
-    // 6. Return ℤ(mv).
+    let mv = argument.as_str(agent).trim_matches(is_trimmable_whitespace);
+
+    // Shortcut if empty
+    if mv.is_empty() {
+        return Some(BigInt::from(0));
+    }
+
+    // Check for bases
+    let mut s = mv.bytes();
+    let base = match (s.next(), s.next()) {
+        (Some(b'0'), Some(b'b' | b'B')) => Some(2),
+        (Some(b'0'), Some(b'o' | b'O')) => Some(8),
+        (Some(b'0'), Some(b'x' | b'X')) => Some(16),
+        _ => None,
+    };
+
+    let string_to_convert = if base.is_some() { &mv[2..] } else { mv };
+    let num_big_int =
+        num_bigint::BigInt::parse_bytes(string_to_convert.as_bytes(), base.unwrap_or(10));
 
-    todo!("string_to_big_int: Implement BigInts")
+    // 6. Return ℤ(mv).
+    Some(BigInt::from_num_bigint(agent, num_big_int.unwrap()))
```

Comment:
question: Is this unwrap okay? I have seen it used in other functions but doesn't feel great in this function. Should I be return `None` if we fail to parse here?